tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/FunExceptRule.scala

```diff
@@ -128,7 +128,8 @@ class FunExceptRule(rewriter: SymbStateRewriter) extends RewritingRule {
       newValue: ArenaCell): SymbState = {
 
     val keyToUpdate = indexEx match {
-      case ValEx(TlaStr(key)) => key
+      case ValEx(TlaStr(key))                           => key
+      case OperEx(TlaFunOper.tuple, ValEx(TlaStr(key))) => key
```

Comment:
@konnov The proved necessary to prevent the rewriter choking on the tuple-wrapped accessor we seem to require. But I dong understand why this rewriting case is "missing" or how the other one can be correct given the apparent requirement for the accessor to be wrapped in a tuple.

Is this rewriting rule inconsistent with the requirements we have elsewhere that this be wrapped in a tuple, or am I missing something about the way these constructions are treated?

Reply:
As far as I can tell, this code applies to the case of the expression being a record. I don't think that you have to match the tuple case, as `indexEx` is already unwrapped in the caller:

https://github.com/informalsystems/apalache/blob/ca4e917e0785d8f82a28febedcb7d849f7fa829a/tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/FunExceptRule.scala#L36

If a record update arrives here as `tuple(tuple(string))`, then something wrong happened during its construction.